lib/forms/pegasus_form_validation.rb

```diff
@@ -136,4 +136,30 @@ def validate_form(kind, data, logger = nil)
 
     data
   end
+
+  # Recursively replace special characters with HTML entities for string content in the input.
+  # This function can be called before saving user input to database or before returning user input
+  # from database to client for presentation.
```

Comment:
You mention this can be used before saving to the database or returning a result. I wonder if it's worth mentioning that doing _both_ would be bad: This function isn't idempotent, and if you called it in both places you would end up passing malformed contents back to the client.

```
escape_html_entities(escape_html_entities('&')) => '&amp;amp;'
```

Reply:
Oh good point. I'll update the comment.